clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/CheckAndUpdatePositionsEvent.java

```diff
@@ -32,4 +32,8 @@ public class CheckAndUpdatePositionsEvent extends CompletableApplicationEvent<Bo
     public CheckAndUpdatePositionsEvent(long deadlineMs) {
         super(Type.CHECK_AND_UPDATE_POSITIONS, deadlineMs);
     }
+
+    public CheckAndUpdatePositionsEvent(long deadlineMs, boolean isPassedByErrorEvent) {
+        super(Type.CHECK_AND_UPDATE_POSITIONS, deadlineMs, isPassedByErrorEvent);
+    }
```

Comment:
At this point it looks like `CheckAndUpdatePositionsEvent` is the only event that uses `isPassedByErrorEvent`. Any reason not to define it in this event as opposed to the superclass?

Reply:
For me, a `CompletableApplicationEvent` is an event that needs to be handled by the application thread. The application thread has two options for processing such events: `processBackgroundEvent` and `addAndGet`. Therefore, every event should have this parameter. However, currently, only the `CheckAndUpdatePositionsEvent` uses it.